notebooks/03-query/15-pivot-unpivot.ipynb

```diff
@@ -0,0 +1,433 @@
+    "    PIVOT ( <list of aggregations>\n",
+    "            \n",
+    "            FOR <pivot_column> IN (<list of values in pivot_column>)\n",
+    "          )\n",
+    "```\n",
+    "- \\<list of aggregations\\> - list of aggregate expressions, example : SUM(added) as added, SUM(deleted) as deleted \n",
+    "- \\<pivot_column\\> - the column whose values are being turned into new columns\n",
+    "- \\<list of values in pivot_column\\> to use in the form `<literal value> as <value_name>,...`, Example: 'true' as robot, 'false as human \n",
+    "\n",
+    "The operation produces columns in the form \\<value_name\\>__\\<aggregation_name\\> for all the aggregations and values that are specified.\n",
+    "\n",
+    "Try this out in the next two cells, first without PIVOT and then with PIVOT to demonstrate what it does:            "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0b36cb33-1e8a-4284-966c-a3937c8a33a4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "    SELECT\n",
+    "      \"channel\", \"isRobot\", \"added\", \"deleted\"\n",
+    "    FROM \"example-wiki-pivot-unpivot\" \n",
+    "    WHERE TIME_IN_INTERVAL(__time, '2016-06-27/P1D')\n",
+    "    LIMIT 10\n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e4a63419-a824-405c-90c2-524611c4024b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "    SELECT\n",
+    "      \"channel\", \n",
+    "       \"robot_added\", \n",
+    "       \"human_added\", \n",
+    "       \"robot_deleted\", \n",
+    "       \"human_deleted\"\n",
+    "    FROM \"example-wiki-pivot-unpivot\" \n",
+    "    \n",
+    "    PIVOT ( SUM(added) as added, \n",
+    "            SUM(deleted) as deleted \n",
+    "            \n",
+    "            FOR \"isRobot\" IN ('true' as robot, 'false' as human)\n",
+    "          )\n",
+    "    WHERE TIME_IN_INTERVAL(__time, '2016-06-27/P1D')\n",
+    "    LIMIT 10\n",
+    "'''\n",
+    "display_client.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0fc3a7ed-255e-49a5-ab0c-964e4fd38885",
+   "metadata": {},
+   "source": [
+    "Notice that the PIVOT operation:\n",
+    "- moved the values of \"isRobot\"='true' into two columns called \"robot_added\" and \"robot_deleted\"\n",
+    "- moved the values of \"isRobot\"='false' into two columns called \"human_added\" and \"human_deleted\"\n",
+    "\n",
+    "Also, notice the presence of NULL values where the pivot column is not the value assigned to the corresponding column name. For example, \"robot_added\" only has values where \"isRobot\" was true."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "841a8fc2-2efa-4327-b35c-e64a8fa380d3",
+   "metadata": {},
+   "source": [
+    "## Robotic updates to wikipedia by channel using normal aggregation\n",
+    "\n",
+    "This is an example of where PIVOT is useful.\n",
```

Comment:
This makes it sound like PIVOT is generally not useful :) 

"The following example demonstrates ..."

Reply:
done